dialog/file.go

```diff
@@ -151,54 +176,100 @@ func (f *fileDialog) makeUI() fyne.CanvasObject {
 	body := fyne.NewContainerWithLayout(layout.NewBorderLayout(scrollBread, nil, nil, nil),
 		scrollBread, f.fileScroll)
 	header := widget.NewLabelWithStyle(label+" File", fyne.TextAlignLeading, fyne.TextStyle{Bold: true})
-	favorites := widget.NewGroup("Favorites", f.loadFavorites()...)
-	return fyne.NewContainerWithLayout(layout.NewBorderLayout(header, footer, favorites, nil),
-		favorites, header, footer, body)
+
+	favorites, err := f.loadFavorites()
+	if err != nil {
+		// only generate the Favorites group if we were able to load
+		// them successfully
+		favorites = []fyne.CanvasObject{}
+		fyne.LogError("Unable to load favorites", err)
+	}
+
+	favoritesGroup := widget.NewGroup("Favorites", favorites...)
+	return fyne.NewContainerWithLayout(layout.NewBorderLayout(header, footer, favoritesGroup, nil),
+		favoritesGroup, header, footer, body)
+
 }
 
-func (f *fileDialog) loadFavorites() []fyne.CanvasObject {
-	home, _ := os.UserHomeDir()
+func (f *fileDialog) loadFavorites() ([]fyne.CanvasObject, error) {
+	osHome, err := os.UserHomeDir()
+	if err != nil {
+		return nil, err
```

Comment:
For this and all those below I don't think we should bomb out if one fails - let's add the favourites that we can and return the last error along with what we build...

Reply:
I rewrote the function. Now it always runs until the end and returns the last error.